### DISCUSSION TITLE
Generating key-val enums from a Graphene API

### DISCUSSION THREAD

**Discussion by jdgreenberger** [dotansimha/graphql-code-generator#4685] [General] [answered]
I'm not sure if this is an issue with graphene or with codegen, but I am unable to generate key-val enums as described in [this issue](https://github.com/dotansimha/graphql-code-generator/issues/3132).

No matter which configuration of graphene.Enum I use, the codegen always creates an enum with the key as the value. Would appreciate any suggestion on how to resolve.

Example
```
class Color(graphene.Enum):
    R = 'red'
    G = 'green'
```
generates 
```
enum ProductType {
   R = 'R'
   G = 'G'
}
```
when I want
```
enum ProductType {
   R = 'red'
   G = 'green'
}
```

**Comment by ardatan** [comment]
Have you tried `enumValues` option?
https://graphql-code-generator.com/docs/plugins/typescript

**Comment by jdgreenberger** [comment]
@ardatan my understanding is that option provides you the ability to override enum values in your config, which would mean copying all of my key-value pairs from the backend. I'm looking for a way, as described in the linked issue, for codegen to automatically create the key-val mappings.

**Reply by ardatan** [reply]
It is not possible to get enum values from an introspection because it is not part of introspection or SDL representation of GraphQL Schema. The only way to get enum values is to use exported JS schema object as explained in that issue.
https://github.com/dotansimha/graphql-code-generator/issues/3132#issuecomment-568794138

**Reply by jdgreenberger** [reply]
I'm not familiar with the exported JS Schema. Does that mean codegen is inspecting a JS file generated by the backend? Do you know if there would be an equivelant for Graphene?

**Reply by ardatan** [reply]
The only way to extract enum values is exported GraphQLSchema generated inside JS backend. Other than that, it is not possible to get those values from your backend.

**Comment by dotansimha** [comment, marked as answer]
As @ardatan said, with the latest versions of codegen, if you are using a code file to load your schema, it will use the enum values from there. 
It's not possible to do that with GraphQL SDL since it doesn't support representing enum values.
